app/src/org/commcare/views/QuestionsView.java

```diff
@@ -320,10 +322,42 @@ private void addHintText(String hintText) {
         }
     }
 
-    public void setFocus(Context context) {
-        if (widgets.size() > 0) {
-            widgets.get(0).setFocus(context);
+    public void setFocus(Context context, int indexOfLastChangedWidget) {
+        QuestionWidget widgetToFocus = null;
+        if (indexOfLastChangedWidget != -1) {
+            widgetToFocus = widgets.get(indexOfLastChangedWidget);
+        } else if (widgets.size() > 0) {
+            widgetToFocus = widgets.get(0);
         }
+        if (widgetToFocus != null) {
+            scrollToWidget(widgetToFocus);
+            widgetToFocus.setFocus(context);
+        }
+    }
+
+    private void scrollToWidget(final QuestionWidget widget) {
+        new Handler().post(new Runnable() {
+            @Override
+            public void run() {
+                QuestionsView.this.scrollTo(0, widget.getTop());
```

Comment:
Think it's an open question whether it's best to scroll to the top of a widget or the bottom (or somewhere in between?). Input from others would be great.